dev/TestMailer.php

```diff
@@ -4,14 +4,14 @@
  * @subpackage email
  */
 class TestMailer extends Mailer {
-	protected $emailsSent = array();
+	protected static $emailsSent = array();
```

Comment:
Ok, the issue wasn't related to `FunctionalTest::post` at all (I thought it might have something to do with Injector nesting that happens within `Director::test`).

The `SapphireTest->mailer` instance gets already overridden/canceled-out within the `setUp` method whenever `self::create_temp_db()` is being called (which in turn calls `SapphireTest::resetDBSchema` that clears the service cache, via `Injector::unregisterAllObjects`).

Solving the problem is relatively simple, by just setting up the Mailer at the end of the `setUp` method. Will amend this PR with these changes and revert the changes made to `TestMailer` (eg. not using static members).